presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -255,6 +256,27 @@ public PlanNode visitSemiJoin(SemiJoinNode node, RewriteContext<Set<Symbol>> con
                     node.getDistributionType());
         }
 
+        @Override
+        public PlanNode visitSpatialJoin(SpatialJoinNode node, RewriteContext<Set<Symbol>> context)
+        {
+            Set<Symbol> expectedFilterInputs = ImmutableSet.<Symbol>builder()
```

Comment:
This is misnamed. It's not just the inputs needed by the filter, but what the downstream node expects. I'd just call it `requiredInputs`.